lib/puppet/provider/registry_value/registry.rb

```diff
@@ -100,8 +100,14 @@ def regvalue
           if Puppet::Util::Windows::Registry.RegQueryValueExW(reg.hkey, valuename_ptr,
             FFI::MemoryPointer::NULL, FFI::MemoryPointer::NULL,
             FFI::MemoryPointer::NULL, FFI::MemoryPointer::NULL) == 0
-            # Note - This actually calls read from Win32::Registry not Puppet::Util::Windows::Registry
-            @regvalue[:type], @regvalue[:data] = from_native(reg.read(valuename))
+            contents = ''
```

Comment:
Not sure how I feel about the default being an empty string.  This means an empty string is now either a) an empty and valid byte sequence in the registry or b) any valid byte sequence of arbitrary length.

Currently can't be nil either as the `data` method above will make it think the value is absent.

Reply:
Current behaviour is to error.

This PR will cause resource to indicate both Type and Data to be unset.

I had a stab at making changes [here](https://github.com/TheRealEdwardCullen/puppetlabs-registry/blob/master/lib/puppet/provider/registry_value/registry.rb), which will result in only the Data being shown as incorrect (unset).

If there's some way to more directly indicate that the value could not be read, that would be good. What about simply having the string result be set to "<< Error reading value >>" or similar? There must be **some** string value that would be so unlikely in the real world as to be improbable... What about adding a GUID - that should cause the string to be unique?

I also added additional debugging and changed the format of the error messages slightly (first time in Ruby/Puppet, so not sure how useful they are).